### PR TITLE
feat(gitlab): handle merge results pipelines (#17440)

### LINKED ISSUE #17440: Support GitLab merged results pipelines feature
### What would you like Renovate to be able to do?

Use auto merge while gitlab merged results pipelines is enabled

This is a premium feature https://docs.gitlab.com/ee/ci/pipelines/merged_results_pipelines.html

### If you have any ideas on how this should be implemented, please tell us here.

With merged results pipelines enabled the branch status is empty, which is interpreted as a pending state, so the MR is never merged.

In that case we could also check the MR latest pipeline status.

### Is this a feature you are interested in implementing yourself?

Yes

### PR DESCRIPTION
**Pull request by gbleu**
## Changes

Also check merge request status when checking if merge request can be merged on gitlab

## Context

- Closes #17440 

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [x] No documentation update is required

## How I've tested my work (please tick one)

I have verified these changes via:

- [ ] Code inspection only, or
- [ ] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [x] Both unit tests + ran on a real repository

### CODE REVIEW

File: lib/modules/platform/gitlab/types.ts [hunk 1]
```diff
@@ -30,6 +30,9 @@ export interface GitLabMergeRequest {
   reviewers?: GitLabUser[];
   labels: string[];
   sha: string;
+  head_pipeline: {
```
Comment (viceice) [inline lib/modules/platform/gitlab/types.ts:33]:
```suggestion
  head_pipeline?: {
```

File: lib/modules/platform/gitlab/index.ts [hunk 2]
```diff
@@ -606,18 +611,21 @@ export async function createPr({
...
-export async function getPr(iid: number): Promise<Pr> {
+export async function getPr(
+  iid: number
+): Promise<Pr & { headPipelineStatus?: string }> {
   logger.debug(`getPr(${iid})`);
   const mr = await getMR(config.repository, iid);
 
   // Harmonize fields with GitHub
-  const pr: Pr = {
+  const pr: Pr & { headPipelineStatus?: string } = {
```
Comment (viceice) [inline lib/modules/platform/gitlab/index.ts:621]: create a gitlab PR interface and extend normal PR instead

**Review by viceice — CHANGES_REQUESTED** [review]

File: lib/modules/platform/gitlab/index.ts [hunk 3]
```diff
@@ -355,6 +355,10 @@ interface GitlabBranchStatus {
   allow_failure?: boolean;
 }
 
+interface GitlabPr extends Pr {
+  headPipelineStatus?: string;
+}
+
```
Comment (viceice) [inline lib/modules/platform/gitlab/index.ts:361]: move to `./types.ts`

File: lib/modules/platform/gitlab/index.ts [hunk 4]
```diff
@@ -414,6 +418,11 @@ export async function getBranchStatus(
     return BranchStatus.yellow;
   }
   logger.debug(`Got res with ${branchStatuses.length} results`);
+
+  const mrStatus = (await getBranchPr(branchName))?.headPipelineStatus;
+  if (!is.undefined(mrStatus)) {
+    branchStatuses.push({ status: mrStatus as BranchState, name: '' });
```
Comment (viceice) [inline lib/modules/platform/gitlab/index.ts:424]: doesn't the head pipeline object also has a name, wich can be used here? if not use a synthetic name like `head_pipeline` or something like that.

Comment (gbleu) [reply lib/modules/platform/gitlab/index.ts]: Does not seems so https://docs.gitlab.com/ee/api/merge_requests.html#get-single-mr

**Review by rarkins — APPROVED** [review]

**Review by viceice — APPROVED** [review]

**Pull request merged**